src/components/MainNav/MyProfile/MyProfile.js

```diff
@@ -47,6 +55,32 @@ export default class MyProfile extends Component {
     this.getProfileImage = this.getProfileImage.bind(this);
   }
 
+  componentDidMount() {
+    const links = this.props.modules.settings.map(setting => setting.links);
+
+    if (links && Array.isArray(links)) {
+      const linkItems = links[0].userDropdown;
+
+      linkItems.forEach((item, index) => {
+        this.isShownToUser().then((response) => {
```

Comment:
Woah -- we're doing an HTTP request for each item in the menu, every time we render the menu? That _can't_ be right. Dump all this stuff, and simply only add the contents of the `links` array if the user identity is already known.

Reply:
... i.e. if `this.props.stripes.user` is defined.